proto/sedachain/batching/v1/batching.proto

```diff
@@ -29,27 +28,24 @@ message Batch {
   bytes proving_metadata = 7;
 }
 
-// TreeEntries are the given batch's data result tree entries and
-// validator tree entries.
-message TreeEntries {
-  // batch_number is the identifier of the batch.
-  uint64 batch_number = 1;
-  // data_result_entries are the entries (unhashed leaf contents) of
-  // the data result tree.
-  repeated bytes data_result_entries = 2;
-  // validator_entries are the entries (unhashed leaf contents) of
-  // the validator tree.
-  repeated bytes validator_entries = 3;
+// DataResultTreeEntries is a list of data result tree entries for a
+// given batch.
+message DataResultTreeEntries { repeated bytes entries = 1; }
+
+// ValidatorTreeEntry is an entry in the validator tree.
+message ValidatorTreeEntry {
+  bytes validator_address = 1
+      [ (gogoproto.casttype) =
+            "github.com/cosmos/cosmos-sdk/types.ValAddress" ];
+  uint32 voting_power_percent = 2;
+  Secp256k1Entry secp256k1 = 3 [ (gogoproto.nullable) = false ];
 }
 
-// BatchSignatures contains basic validator data and its batch signatures
-// under various cryptographic schemes.
-message BatchSignatures {
-  // batch_number is the identifier of the batch.
-  uint64 batch_number = 1;
-  string validator_addr = 2
-      [ (cosmos_proto.scalar) = "cosmos.ValidatorAddressString" ];
-  bytes signatures = 3;
+// Secp256k1Entry is the secp256k1 signature component of a validator
+// tree entry.
+message Secp256k1Entry {
+  bytes eth_address = 1;
```

Comment:
It doesn't have to be an eth address right? Could be any evm network, and even then I don't think it's limited to only evm networks.

Reply:
Right but I think we will have to create a new struct every time we add a new signature scheme

Reply:
`eth_address` means more like ethereum-style address format. So different chains could still use this if they use this address format.